### PR TITLE
Add tests for async example.

### PR DESCRIPTION
**Pull request by jonboiser**
Per #346 

Let me know if there are any issues. I'll look into the 'real-world' example tomorrow; hopefully it's just as straightforward.

### CODE REVIEW

File: examples/async/test/sagas.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    generator.next().value,
+    call(fetchPostsApi, redditName),
+    'must call fetchPostsApi with reddit name'
+  )
+  t.deepEqual(
+    generator.next(mockPosts).value,
+    put(actions.receivePosts(redditName, mockPosts)),
+    'must dispatch a receivePosts action with posts'
+  )
+  t.equal(generator.next().value, undefined, 'must finish')
```
Comment (yelouafi) [inline examples/async/test/sagas.js]: I think `t.ok(generator.next().done, 'must finish')` conveys better meaning.

**Comment by jonboiser** [comment]
Made the change. Not sure were I got that pattern from, might have been a blog post.

**Comment by yelouafi** [comment]
The rest seems Ok. So I'll merge this.

Thanks for the PR!

**Comment by jonboiser** [comment]
No problem! I'll see if I can tackle the 'real-world' example soon.

**Pull request merged**